lib/otw_sanitize/media_sanitizer.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+# Creates a Sanitize transformer to sanitize audio and video tags
+module OTWSanitize
+  class MediaSanitizer
+    # Attribute whitelists
+    AUDIO_ATTRIBUTES = %w[
+      class controls crossorigin dir
+      loop muted preload src title
+    ].freeze
+
+    VIDEO_ATTRIBUTES = %w[
+      class controls crossorigin dir height loop
+      muted playsinline poster preload src title width
+    ].freeze
+
+    SOURCE_ATTRIBUTES = %w[src type].freeze
+    TRACK_ATTRIBUTES = %w[default kind label src srclang].freeze
+
+    WHITELIST_CONFIG = {
+      elements: %w[audio video source track],
+      attributes: {
+        'audio'  => AUDIO_ATTRIBUTES,
+        'video'  => VIDEO_ATTRIBUTES,
+        'source' => SOURCE_ATTRIBUTES,
+        'track'  => TRACK_ATTRIBUTES
+      },
+      add_attributes: {
+        'audio' => {
+          'controls'    => 'controls',
+          'crossorigin' => 'anonymous',
+          'preload'     => 'metadata'
+        },
+        'video' => {
+          'controls'    => 'controls',
+          'playsinline' => 'playsinline',
+          'crossorigin' => 'anonymous',
+          'preload'     => 'metadata'
+        }
+      },
+      protocols: {
+        'video' => {
+          'poster' => %w[http https]
```

Comment:
Why do we need to specify the allowed protocols for `poster` but not `src`?

Reply:
It prevents you from putting scripts in, which was part of the spec, but I can add it to source too.